google/cloud/spanner_dbapi/cursor.py

```diff
@@ -258,9 +260,56 @@ def executemany(self, operation, seq_of_params):
 
         many_result_set = StreamedManyResultSets()
 
-        for params in seq_of_params:
-            self.execute(operation, params)
-            many_result_set.add_iter(self._itr)
+        if classification in (parse_utils.STMT_INSERT, parse_utils.STMT_UPDATING):
+            statements = []
+
+            for params in seq_of_params:
+                sql, params = parse_utils.sql_pyformat_args_to_spanner(
+                    operation, params
+                )
+                statements.append((sql, params, get_param_types(params)))
+
+            if self.connection.autocommit:
+                transaction = self.connection.transaction_checkout()
```

Comment:
I think it would be better to handle this in the same way as that we are handling single statements in autocommit, which means just using the `run_in_transaction` function in the client library: https://github.com/googleapis/python-spanner/blob/62b937d6889fc56a64e5073985bab8ccd5d8c24d/google/cloud/spanner_dbapi/cursor.py#L232

Reply:
I had a thought about it, but it will not give anything, you know, significant, while testing will become more complex (because transactions will be initiated deeper in the API). Also it'll be little bit harder to get part of the result in case of an error failed during the batch execution.

So, it looks like a change to the same.

Reply:
I don't agree that it is the same. If you use `run_in_transaction`, aborted transactions will automatically be retried by the client library. In this implementation any aborted transaction will be propagated as an error to the client application. That means that if you use `execute` in autocommit mode, statements will automatically be retried if the underlying transaction is aborted, while if you use `execute_many` in autocommit mode, the aborted error will be propagated to the client application. I don't think that is a very good API.
The other option would be to use the checksum retry logic for autocommit as well, but that seems like over-complicating things to me.